src/lib.rs

```diff
@@ -13,114 +13,90 @@ pub mod iso_specs;
 pub mod pds;
 
 use crate::iso_specs::Category;
-use crate::iso_field::FieldCharType;
 use eyre::{eyre, Result};
 use std::collections::HashMap;
 use std::fmt;
 use strum::{EnumProperty, IntoEnumIterator};
 
 #[derive(Debug, Clone, Serialize)]
-pub struct Message {
-    label: String,
-    value: Vec<u8>,
-    char_type: FieldCharType,
+pub struct Field {
+    pub label: String,
+    pub value: Vec<u8>,
 }
 
-impl Message {
+impl Field {
     pub fn utf8_value(&self) -> String {
         String::from_utf8_lossy(&self.value).to_string()
     }
 
-    pub fn get_value(&self) -> &Vec<u8> {
-        &self.value
-    }
-
-    pub fn get_char_type(&self) -> &FieldCharType {
-        &self.char_type
-    }
-
     pub fn get_label(&self) -> String {
         self.label.to_string()
     }
 }
 
-impl fmt::Display for Message {
+impl fmt::Display for Field {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match std::str::from_utf8(&self.value) {
-            Ok(message_value) => write!(f, "{}: {}", self.label, message_value),
+            Ok(field_value) => write!(f, "{}: {}", self.label, field_value),
             Err(_) => write!(f, "{}: {:02X?}", self.label, self.value),
         }
     }
 }
 
-/// A Group represents a set of messages e.g Data elements or PDS
-/// Data elements (DE) are stored within `messages`
+/// A Message represents a set of fields e.g Data elements or PDS
 ///
-/// Usually a group represents something based on it's categories, for example a FirstPresentment
+/// Usually a message represents something based on it's categories, for example a FirstPresentment
 /// Although some messages rely on being chained, like a MessageException, linked to a FirstPresentment on a TT113 file
 #[derive(Debug, Clone, Serialize)]
-pub struct Group {
-    //FIXME this could be a hashmap just like pds, and also named to DE
-    pub messages: Vec<Message>,
-    //FIXME for now pds are only implemented for when de48 is present
-    pub pds: HashMap<String, String>,
+pub struct Message {
+    pub mti: String,
     pub category: Category,
+    pub primary_bitmap: [u8; 8],
+    pub data_elements: HashMap<String, iso_field::IPMValue>,
+    pub pds: HashMap<String, String>,
 }
 
-impl Group {
-    /// Returns a HashMap [message.label => message.utf8_value] of all self.messages
+impl Message {
+    /// Returns a HashMap [message.label_id => message.utf8_value] of all self.messages
     pub fn get_messages_hash(&self) -> Result<HashMap<String, String>> {
```

Comment:
Considering that is a legacy code, let's remove it.

Reply:
`get_messages_hash`  is a method I created to use here at Receivables, we need this method. Why do you guys wanna remove it? Always remember that we use this lib too.